src/gui/wizard/owncloudwizard.cpp

```diff
@@ -104,6 +112,33 @@ OwncloudWizard::OwncloudWizard(QWidget *parent)
 
     // allow Flow2 page to poll on window activation
     connect(this, &OwncloudWizard::onActivate, _flow2CredsPage, &Flow2AuthCredsPage::slotPollNow);
+
+    adjustWizardSize();
+}
+
+void OwncloudWizard::adjustWizardSize()
+{
+    QList<QSize> pageSizes;
+    const auto pIds = pageIds();
+    std::transform(pIds.begin(), pIds.end(), std::back_inserter(pageSizes), [this](int pageId) {
+        auto p = page(pageId);
+
+        p->adjustSize();
+
+        return p->sizeHint();
+    });
+
+    const auto maxPageSize = *std::max_element(pageSizes.begin(), pageSizes.end(), [](const QSize &size1, const QSize &size2) {
+        if (size1.width() < size2.width()) {
+            return true;
+        }
+        if (size1.height() < size2.height()) {
+            return true;
+        }
+        return false;
+    });
```

Comment:
Now I realize that using `max_element` was probably not the best idea... it's more std::accumulate we're after to have a size with the maximum width and height. Otherwise you might end up with the largest width or the largest height in the end but not both (depends on the ordering in the list).